From 31d801cf4ea2e7e7df3090b63aa91de4cc428541 Mon Sep 17 00:00:00 2001
From: ahs <amrita.h.s@intel.com>
Date: Wed, 13 Jan 2021 17:45:41 +0530
Subject: [PATCH 3/4] Disable avx versions of sinf, cosf and expf functions

AVX2 implementation of these APIs is causing
CtsGraphicsTestCases failure

Change-Id: Ib812c70150bf6f650e6666e5a37660c39b5349a1
Tracked-On: OAM-95862
Signed-off-by: ahs <amrita.h.s@intel.com>
---
 libm/x86_64/dynamic_function_dispatch.cpp | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/libm/x86_64/dynamic_function_dispatch.cpp b/libm/x86_64/dynamic_function_dispatch.cpp
index 3710873a9..94f6a3f1c 100644
--- a/libm/x86_64/dynamic_function_dispatch.cpp
+++ b/libm/x86_64/dynamic_function_dispatch.cpp
@@ -132,8 +132,8 @@ DEFINE_IFUNC_FOR(cos) {
 
 typedef float cosf_func(float __x);
 DEFINE_IFUNC_FOR(cosf) {
-    __builtin_cpu_init();
-    if (__builtin_cpu_supports("avx2")) RETURN_FUNC(cosf_func, cosf_avx2);
+    //__builtin_cpu_init();
+    //if (__builtin_cpu_supports("avx2")) RETURN_FUNC(cosf_func, cosf_avx2);
     RETURN_FUNC(cosf_func, cosf_generic);
 }
 
@@ -146,8 +146,8 @@ DEFINE_IFUNC_FOR(sin) {
 
 typedef float sinf_func(float __x);
 DEFINE_IFUNC_FOR(sinf) {
-    __builtin_cpu_init();
-    if (__builtin_cpu_supports("avx2")) RETURN_FUNC(sinf_func, sinf_avx2);
+    //__builtin_cpu_init();
+    //if (__builtin_cpu_supports("avx2")) RETURN_FUNC(sinf_func, sinf_avx2);
     RETURN_FUNC(sinf_func, sinf_generic);
 }
 
@@ -167,8 +167,8 @@ DEFINE_IFUNC_FOR(sinhf) {
 
 typedef float expf_func(float __x);
 DEFINE_IFUNC_FOR(expf) {
-    __builtin_cpu_init();
-    if (__builtin_cpu_supports("avx2")) RETURN_FUNC(expf_func, expf_avx2);
+    //__builtin_cpu_init();
+    //if (__builtin_cpu_supports("avx2")) RETURN_FUNC(expf_func, expf_avx2);
     RETURN_FUNC(expf_func, expf_generic);
 }
 
-- 
2.17.1

